File: 02.Data_Files/readAndParseFiles.js
import fs from "fs";
import yaml from "js-yaml";
import { XMLParser } from "fast-xml-parser";
import csv from "csv-parser";

export function readJSON(filePath) {
  const fileContent = fs.readFileSync(filePath, "utf-8");
  return JSON.parse(fileContent);
}

export function readYAML(filePath) {
  const fileContent = fs.readFileSync(filePath, "utf-8");
  return yaml.load(fileContent);
}

export function readXML(filePath) {
  const fileContent = fs.readFileSync(filePath, "utf-8");
  const parser = new XMLParser();
  return parser.parse(fileContent);
}

export async function readCSV(filePath) {
  const results = [];
  const stream = fs.createReadStream(filePath).pipe(csv());

  for await (const data of stream) {
    results.push(data);
  }

  return results;
}

export function readTextFile(filePath) {
  const fileContent = fs.readFileSync(filePath, "utf-8");
  return { text: fileContent };
}

File: 02.Data_Files/readAndParseFiles.py
# XML ----------------------------------------------------------------------------------------
import xml.etree.ElementTree as ET


tree = ET.parse("me.xml")
root = tree.getroot()
print("XML Data:")
for child in root:
    print(child.tag, child.text)
    if len(list(child)) > 0:
        for subChild in child:
            print(subChild.tag, subChild.text)
            tree = ET.parse("me.xml")


# CSV ----------------------------------------------------------------------------------------
import csv

with open("me.csv", newline="") as csvFile:
    reader = csv.reader(csvFile)
    print("CSV Data:")
    for row in reader:

        print(row)

# YAML ----------------------------------------------------------------------------------------

import yaml

with open("me.yaml", "r") as yamlFile:
    data = yaml.safe_load(yamlFile)
    print("YAML Data:")
    print(data)


# JSON  ----------------------------------------------------------------------------------------
import json

with open("me.json", "r") as jsonFile:
    data = json.load(jsonFile)
    print("JSON Data:")
    print(data)

# Text ----------------------------------------------------------------------------------------

file_path = "me.txt"

with open(file_path, "r") as file:
    file_content = file.read()

dict_data = {"text": file_content}

print(dict_data)
